pkg/kubeapiserver/server/insecure_handler.go

```diff
@@ -111,3 +113,15 @@ func serveInsecurely(insecureServingInfo *InsecureServingInfo, insecureHandler h
 	_, err = server.RunServer(insecureServer, insecureServingInfo.BindNetwork, stopCh)
 	return err
 }
+
+// insecureSuperuser implements authenticator.Request to always return a superuser.
+// This is functionally equivalent to skipping authentication and authorization,
+// but allows apiserver code to stop special-casing a nil user to skip authorization checks.
+type insecureSuperuser struct{}
+
+func (insecureSuperuser) AuthenticateRequest(req *http.Request) (user.Info, bool, error) {
+	return &user.DefaultInfo{
+		Name:   "system:unsecured",
+		Groups: []string{user.SystemPrivilegedGroup, user.AllAuthenticated},
```

Comment:
Thanks for the explanation. I noticed it while reading through the audit logs of an e2e test, and it struck me as odd.

```
2017-06-02T02:21:03.555917358Z AUDIT: id="286a4484-480d-41b8-b9b4-4c30305fe0f3" stage="ResponseComplete" ip="127.0.0.1" method="get" user="system:unsecured" groups="\"system:masters\",\"system:authenticated\"" as="<self>" asgroups="<lookup>" namespace="kube-system" uri="/api/v1/namespaces/kube-system/configmaps/ingress-uid" response="200"
```